modules/nf-core/samtools/collatefastq/main.nf

```diff
@@ -23,21 +25,27 @@ process SAMTOOLS_COLLATEFASTQ {
     def args = task.ext.args ?: ''
     def args2 = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def reference = fasta ? "--reference ${fasta}" : ""
+    def output = interleave ? "> ${prefix}_interleaved.fq.gz" : "-1 ${prefix}_1.fq.gz -2 ${prefix}_2.fq.gz -s ${prefix}_singleton.fq.gz"
+
     """
     samtools collate \\
         $args \\
         --threads $task.cpus \\
+        ${reference} \\
         -O \\
         $input \\
         . |
 
     samtools fastq \\
         $args2 \\
         --threads $task.cpus \\
-        -1 ${prefix}_1.fq.gz \\
-        -2 ${prefix}_2.fq.gz \\
+        ${reference} \\
         -0 ${prefix}_other.fq.gz \\
-        -s ${prefix}_singleton.fq.gz
+        $output
+
+    # touch file so the outputs work
+    touch ${prefix}_singleton.fq.gz
```

Comment:
Works for me, I only wanted to avoid splitting everything into separate channels and keep the outputs as is.
What do you suggest? Separate channel for split, interleaved, singleton and other?

Reply:
Yes I think I would personally prefer that personally. I generally agree avoiding splitting is nice, but not if it requires maing 'fake' output

Reply:
I changed the outputs, but on second thought, I could've just said the singleton was `optional`. Never even crossed my mind 😓